snx-rs: 2.2.3 -> 2.9.0

Shavyn opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

This is only a version bump of the snx-rs package as the version in nixpkgs (2.2.3) didn't support a feature added later I needed.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nbraud commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 376344`

nbraud commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 376344`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>snx-rs</li>
  </ul>
</details>

nicoonoclaste reviewed (approved):
Seems good.

nbraud commented:
@Shavyn This package is currently unmaintained (`meta.maintainers == []`) so it might not be kept up-to-date  :grimacing: 
Since you are using it, would you volunteer for maintainership?

I'd also suggest setting up [automatic updates](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#automatic-package-updates) so @r-ryantm will automatically generate PRs and notify the maintainer(s)

Adda0 reviewed (commented):
Before merging, the commit history should be definitely cleaned up: squashing the commits, removing the `merged from ...` commits, ideally keeping only one commit, which can then have additional information in the commit long description part from the current commits, if you want.

Adda0 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 376344`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>snx-rs</li>
  </ul>
</details>

### Reviewed points
<details>
  <summary>:white_check_mark: 12 | :grey_question: 1 | :red_circle: 1</summary>

  - [x] Package name [fits the guidelines](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming).
  - [x] Package version [fits the guidelines](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning).
  - [ ] The commit text [fits the guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions).
  - [x] Package maintainers are notified.
    - The continuous integration system will make GitHub notify users based on the submitted changes, but it can happen that it misses some of the package maintainers.
  - [x] `meta` field information [fits the guidelines](https://github.com/NixOS/nixpkgs/edit/master/pkgs/README.md#meta-attributes) and contain the correct information:
    - [x] License can change with version updates, so it should be checked to match the upstream license.
    - [x] If the package has no maintainer, a maintainer must be set. This can be the update submitter or a community member that accepts to take maintainership of the package.
  - [x] The code contains no typos.
  - [x] The package builds locally on x86_64-linux.
  - [x] All executables run on x86_64-linux.
  - [ ] All executables tested on x86_64-linux.
  - [x] All depending packages build.
  - [x] Patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed.
  - [x] Patches that are remotely available are fetched rather than vendored.

</details>

##### Possible improvements

##### Comments

Shavyn commented:
@nbraud thanks for the reply. I'll consider adding the automation and the possibility to maintain the package (my knowledge of Nix is next to zero).
@Adda0 thanks for your contribution. As you can guess I'm a newbie and this is all new to me. I'm learning how to clean up this mess.

Adda0 commented:
I looked at the upstream repo. They have a [changelog file](https://github.com/ancwrd1/snx-rs/blob/main/CHANGELOG.md) in there. We should add it to the `meta` attrset like this:
```nix
changelog = "https://github.com/ancwrd1/snx-rs/blob/v${version}/CHANGELOG.md";
``` 

Could you make the change while you are at it, too?

Adda0 commented:
> @nbraud thanks for the reply. I'll consider adding the automation and the possibility to maintain the package (my knowledge of Nix is next to zero).

Since this project has regular releases on GitHub, adding an update script should be as easy as adding
```
passthru.updateScript = nix-update-script { };
```

Adda0 commented on pkgs/by-name/sn/snx-rs/package.nix:
```diff
@@ -1,44 +1,63 @@
+
   checkFlags = [
     "--skip=platform::linux::net::tests::test_default_ip"
+    "--skip=platform::linux::resolver::tests::test_detect_resolver"
```
It would be good to add an explanation (a comment) here for why the test fails.

Is it something that should be reported upstream?

Shavyn replied:
It was already there when I tried to build an older version.

Trying to build the package with that flag commented out before squashing commits and submit it again. If it fails I'll report it upstream.

Thanks for pointing this out.

nbraud replied:
The issue is likely that the build sandbox doesn't have access to the network, FWIW.

Shavyn replied:
Thanks for your help. Fortunately it didn't fail when I commented that line out. So I think this can be resolved. Right?

Adda0 replied:
If it works, then all the better. I consider this conversation to be resolved.

Shavyn commented:
Thanks for all the inputs, as soon as I come back from work I'll have everything (hopefully) sorted.

nbraud commented:
> Before merging, the commit history should be definitely cleaned up: squashing the commits, removing the `merged from ...` commits, ideally keeping only one commit

TBH, that can be accomplished just as well with a squash merge, so I don't see a strong incentive to make people squash the history themselves, especially given that it's then harder for others to see what changed since they last reviewed.

Shavyn commented:
I managed to:
- add changelog to meta informations
- remove the failing test
- (hopefully) squash commits

Sorry it took me so long. For that reason I'm hesitating volunteering to maintain the package. Let me train a little bit more :).

Adda0 commented on pkgs/by-name/sn/snx-rs/package.nix:
```diff
@@ -1,51 +1,74 @@
 
+  postPatch = ''
+    substituteInPlace $cargoDepsCopy/libappindicator-sys-*/src/lib.rs \
+      --replace "libayatana-appindicator3.so.1" "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1"
```
```suggestion
      --replace-fail "libayatana-appindicator3.so.1" "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1"
```
Maybe use `replace-fail` to warn us when something goes awry?

Shavyn replied:
Got it. Updating the branch. Another little thing learned on how Nix works.

Shavyn replied:
The latest push adds what you suggested.

Adda0 commented:
> TBH, that can be accomplished just as well with a squash merge, so I don't see a strong incentive to make people squash the history themselves, especially given that it's then harder for others to see what changed since they last reviewed.

True, but it has disadvantages that, IMO, heavily outweigh its usefulness:
1. This puts the responsibility to the committers to perform another check before merging and carefully think whether squashing the commits makes sense for each PR.
2. This prevents separate meaningful commits when it really makes sense: PRs with changes different from simple updates. Again, the committers must spend more time reviewing each PR.
3. One cannot separate the update to the maintainer list from the update to a package.

Personally, I prefer to make the job of committers as easy as possible, and put the responsibility of preparing the PR to an ideal state to each contributor.

The only real disadvantage I see here is, as you point out, that the diff for the previous reviewers gets mangled. But here we are usually not reviewing "program" code spanning multiple source files, but "only" one "configuration" file which would get modified anyway, so unless you review the specific new commits separately, you will see that the entire file has changed, anyway.

Adda0 commented:
> I managed to:
> 
>     * add changelog to meta informations
> 
>     * remove the test disabling
> 
>     * (hopefully) squash commits
> 
> 
> Sorry it took me so long. For that reason I'm hesitating volunteering to maintain the package. Let me train a little bit more :).

Thanks. It looks great. In general, one should not merge the master branch into the feature branch (creating the merge commit), but instead rebase onto the master branch. This would result in a single commit built atop the newest commit on the master branch.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/packages-looking-for-a-maintainer/5442/9

Shavyn commented:
The latest commit adds what @Adda0 suggested in his review, adds myself as maintainer (wish me good luck). I hope I used the git rebase command correctly instead of merging 

Thanks everyone for your help.

Adda0 reviewed (commented):
Other than the commit for maintainer list update, all seems well. I will do a final rebuild of all binaries soon and if all goes well, I am all for approving the PR. Thank you for the patience with our "demands" and good luck with the maintainership. Hopefully, all will work well with future versions of the package, and there will not be much additional work needed.

And yes, the rebase seems to went well. Great work!

Shavyn commented on maintainers/maintainer-list.nix:
That was one of the (many) things I didn't notice when submitting the PR.

Latest push solves also this, at least I hope so :).

PS: you must be the one with patience with newbies like me.

Adda0 replied:
Thanks.

Just for future reference, the message of the commit should be `maintainers: Add shavyn`, not `maintainers: Add <Shayn>`, the angle brackets from the guide are there only as a placeholder with the name of the placeholder inside the angle brackets where one is expected to replace the whole `<name_of_the_placeholder>` with their text. This is often seen in Linux guides etc. E.g.:
```sh
# Run the following command with your user specified:
sh ./assign-user.sh -u <user> -t <timeout> -o <output>
```
where `<user>` can be, for example, `adam`, `<timeout>` can be `60`, and `<output>` can be `output_file.log`.

Shavyn replied:
Argh!!! Sorry about that. You want me to amend the commit?

Adda0 replied:
No worries. I think it is not that important, unless you really want to practice rebasing :)

Adda0 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 376344`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>snx-rs</li>
  </ul>
</details>

##### Comments

All builds and runs, but I cannot test the functionality of the package. Looks good to me.

Shavyn committed:
maintainers: add shavyn

Shavyn committed:
snx-rs: 2.2.3 -> 2.9.0

https://github.com/ancwrd1/snx-rs/blob/v2.9.0/CHANGELOG.md

Added dependencies for snx-rs GUI to work.

Shavyn commented:
Removed the (unintentional) merge master commit. Sorry.
